pkg/api/customization/monitor/metric_action.go

```diff
@@ -34,6 +34,7 @@ func (h *MetricHandler) Action(actionName string, action *types.Action, apiConte
 		var clusterName, projectName, appName, saNamespace string
 		var comm v3.CommonQueryMetricInput
 		var err error
+		var svcNamespace, svcName string
```

Comment:
do all your action handlers have auth (RBAC) checks on them? i dont see them

Reply:
I believe you are basically saying that we defer RBAC down to the prometheus-auth level. That makes sense.

But, as of now, these actions will show up in the API for all users, regardless of access level, right? So, read-only will see the links, but they won't function properly.

Could we add logic to the formatter (which adds the action links to the API) to only show add these links if the user has the `view` permission on the `prometheus.cattle.io` resource in the project or cluster?

Would that allow the UI appropriately hide/show links to view monitoring? Would it require a UI change?

cc @loganhz @thxCode @orangedeng

Reply:
i may not have the full picture, so please correct if im wrong on anything above

Reply:
can be addressed outside of this PR